lib/puppet_x/puppetlabs/registry.rb

```diff
@@ -172,33 +168,19 @@ def self.combine_path_and_value(keypath, valuename)
 
     # Extract the valuename from the path and then munge the actual path
     def initialize(path)
-      # Try finding the valuename via the double backslash method first
-      # and then revert to the old single backslash way
-      dbl_slash_idx = path.index('\\\\') || 0
-      if dbl_slash_idx > 0
-        # If the user specified a double backslash, split the string there
-        # Strip the valuename from the path
-        @valuename = path[dbl_slash_idx+2..-1]
+      # valuename appears after the the first double backslash
+      path, @valuename = path.split('\\\\', 2)
+      # found \\ so nearly done parsing
+      if !@valuename.nil?
+        @is_default = @valuename.empty?
+      # no \\ but there is at least a single \
+      elsif path.include?('\\')
+        path, _, @valuename = path.rpartition('\\')
         @is_default = @valuename.empty?
-        path = path[0..dbl_slash_idx-1]
+      # a hive name that contains no \
```

Comment:
I replicated the third branch of the existing code logic, but it doesn't make sense to me as we don't actually support something like `"hklm"` as a path without at least one slash like `"hklm\"` (based on the rules in the type).  I'm seeing if we can remove that as well to simplify further.

Reply:
Yup, it can be slightly reworked to simplify things even further.